release/models/bgp/openconfig-bgp-policy.yang

```diff
@@ -661,6 +670,296 @@ module openconfig-bgp-policy {
     }
   }
 
+  grouping ext-community-members-top {
+    description
+      "Top-level grouping for members of an external community set";
+
+    container members {
+        description
+            "Enclosing container for list of extended community members";
+
+        list member {
+            key "index";
+            description
+              "List of members in an external community set.  Each entry
+              is identified by an index.";
+
+          leaf index {
+              type leafref {
+                  path "../config/index";
+              }
+              description
+                  "Reference to the index for the member of an extended
+                  community set.";
+          }
+
+          container config {
+            description "Extended community member configuration.";
+
+            uses ext-community-member-config;
+          }
+
+          container state {
+            config false;
+            description "State for BGP extended community set members";
+
+            uses ext-community-member-config;
+            uses ext-community-member-state;
+          }
+
+          container asn-type {
```

Comment:
Here and following, you have top level containers for communities by their structure.  However, without knowing the type/subtype, you have zero idea what these things are.  For example, route targets can be of these types, but other things may use this structure without being route targets.
My recommendation is that these generic containers are NOT included as they only create ambiguity.

Reply:
The goal by defining this level of containers it to provide a way to identify the 'style' of community being used.  

- asn-type community (2octet ASN rfc4360)
- asn4-type community (4octet ASN rfc5668 section 1) (just added this sub container to clarify)
- ip-type community (rfc5668 section 2)
- link-bw community (draft-ietf-idr-link-bandwidth-07)
- route-target-ip (rfc4360 section 4 using using an IP address)
- route-target (rfc4360 section 4)
- route-origin (rfc4369 section 5)
- route-origin-ip (rfc4369 section 5 using ip address)
- color (rfc9012 and draft-ietf-idr-segment-routing-te-policy)

These containers are similar to using a keyword in a command line community statement.  They go one step further than the example commands below which need to parse strings to derive the 'style'.
From JunOS example has a keyword, but also relies on some string parsing too:
```
set policy-options community origin-as members origin:64511:3
set policy-options community origin-ip members origin:172.16.7.7:4
set policy-options community target-as members target:64511:1
```

These are equivalent to proposed OC:
```
prefix = /routing-policy/defined-sets/bgp-defined-sets/ext-community-sets
$prefix/ext-community-set[name=origin-as]/members/member[index=1]/asn-type/config/asn = "64511"
$prefix/ext-community-set[name=origin-as]/members/member[index=1]/asn-type/config/value = "3"

$prefix/ext-community-set[name=origin-ip]/members/member[index=1]/ip-type/config/asn = "172.16.7.7"
$prefix/ext-community-set[name=origin-ip]/members/member[index=1]/ip-type/config/value = "4"

$prefix/ext-community-set[name=origin-ip]/members/member[index=1]/route-target/config/asn = "64511"
$prefix/ext-community-set[name=origin-ip]/members/member[index=1]/route-target/config/value = "1"
```